app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
+    @user = User.from_omniauth(auth)
+
+    if @user.persisted?
+      flash[:notice] = I18n.t("devise.omniauth_callbacks.success",
+                              kind: "Google")
+      sign_in_and_redirect @user, event: :authentication
+    else
+      session["devise.google_data"] = request.env["omniauth.auth"]
+                                             .except("extra")
+      redirect_to signin_url, alert: @user.errors.full_messages
+                                          .join("\n")
+    end
+  end
+
+  private
+
+  def auth
+    @auth ||= request.env["omniauth.auth"]
```

Comment:
Biến @auth này giá trị lúc em chưa gán là gì nhỉ? Sao lại cần ||= em

Reply:
này e copy trên mạng thấy giờ thấy cũng vô lý để em sửa ạ

Reply:
Assignment Branch Condition size for google_oauth2 is too high
xóa đi bị lỗi này anh ạ